cluster/src/main/java/org/apache/iotdb/cluster/server/heartbeat/HeartbeatThread.java

```diff
@@ -230,10 +235,15 @@ void sendHeartbeatSync(Node node) {
                   heartbeatHandler.onComplete(heartBeatResponse);
                 } catch (TTransportException e) {
                   logger.warn(
-                      "{}: Cannot send heart beat to node {} due to network", memberName, node, e);
+                      memberName
```

Comment:
Although there's nothing wrong with this, I am curious for doing this.

Reply:
Because the `logger.warn` has the following overload signatures with the `Marker` string:

```java
    void warn(Marker var1, String var2);
    void warn(Marker var1, String var2, Object var3);
    void warn(Marker var1, String var2, Object var3, Object var4);
    void warn(Marker var1, String var2, Object... var3);
    void warn(Marker var1, String var2, Throwable var3);
```

We can see that only the last one can print the stacktrace from the last parameter. However, the original code here satisfies `void warn(Marker var1, String var2, Object var3, Object var4);` , 3 parameters with only two placeholders. The execption message and stacktrace are lost.

Reply:
got it